File: python/example/web_development_agent.py
"""
Example: Web Development Agent
Demonstrates WEB_DEV_TOOL for creating and managing web applications.
"""
import asyncio
import os
from baseai import BaseAI
from baseai.tools import BaseAITool


async def main():
    api_key = os.getenv("BASEAI_API_KEY", "YOUR_API_KEY")
    if api_key == "YOUR_API_KEY":
        raise ValueError("Please set BASEAI_API_KEY environment variable or update the api_key in the script")
    client = BaseAI(api_key=api_key, api_url="https://a2abase.ai/api")

    thread = await client.Thread.create()
    desired_name = "Web Development Agent"
    agent = await client.Agent.find_by_name(desired_name)
    created = False
    if agent is None:
        agent = await client.Agent.create(
            name=desired_name,
            system_prompt=(
                "You are a web development assistant specializing in modern web applications. "
                "You can create Next.js applications with shadcn/ui components, set up routing, "
                "create responsive layouts, and implement best practices for web development."
            ),
            mcp_tools=[BaseAITool.WEB_DEV_TOOL],
        )
        created = True

    # Example: Create a modern web application
    run = await agent.run(
        "Create a Next.js application with a landing page that includes a hero section, "
        "features section, and contact form using shadcn/ui components.",
        thread
    )
    stream = await run.get_stream()
    async for line in stream:
        print(line)

    if created:
        await agent.delete()


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/example/web_development_agent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Example: Web Development Agent
 * Demonstrates WEB_DEV_TOOL for creating modern web applications.
 */
const a2abase_1 = require("../src/a2abase");
const tools_1 = require("../src/tools");
async function main() {
    const apiKey = process.env.BASEAI_API_KEY;
    if (!apiKey || apiKey === 'YOUR_API_KEY') {
        throw new Error('Please set BASEAI_API_KEY environment variable');
    }
    const client = new a2abase_1.A2ABase({
        apiKey,
        apiUrl: 'https://a2abase.ai/api',
    });
    const thread = await client.Thread.create();
    const desiredName = 'Web Development Agent';
    let agent = await client.Agent.findByName(desiredName);
    let created = false;
    if (!agent) {
        agent = await client.Agent.create({
            name: desiredName,
            systemPrompt: 'You are a web development assistant. You can create modern web applications using Next.js and shadcn/ui. Help users build responsive, accessible, and performant web applications.',
            a2abaseTools: [tools_1.A2ABaseTool.WEB_DEV_TOOL],
        });
        created = true;
    }
    const run = await agent.run('Create a simple todo list application with Next.js and shadcn/ui components.', thread);
    const stream = await run.getStream();
    for await (const chunk of stream) {
        process.stdout.write(chunk);
    }
    if (created) {
        await agent.delete();
    }
}
main().catch(console.error);
